src/allotropy/parsers/chemometec_nucleoview/nucleoview_parser.py

```diff
@@ -0,0 +1,169 @@
+import io
+import time
+from typing import Any, Optional
+import uuid
+
+import pandas as pd
+
+from allotropy.allotrope.models.cell_counting_benchling_2023_11_cell_counting import (
+    CellCountingAggregateDocument,
+    CellCountingDetectorDeviceControlAggregateDocument,
+    CellCountingDetectorMeasurementDocumentItem,
+    CellCountingDocumentItem,
+    DataProcessingDocument,
+    DataSystemDocument,
+    DeviceControlDocumentItemModel,
+    DeviceSystemDocument,
+    MeasurementAggregateDocument,
+    Model,
+    ProcessedDataAggregateDocument1,
+    ProcessedDataDocumentItem,
+    SampleDocument,
+)
+from allotropy.allotrope.models.shared.definitions.custom import (
+    TQuantityValueCell,
+    TQuantityValueMicrometer,
+    TQuantityValueMillionCellsPerMilliliter,
+    TQuantityValuePercent,
+    TQuantityValueUnitless,
+)
+from allotropy.allotrope.models.shared.definitions.definitions import TDateTimeValue
+from allotropy.constants import ASM_CONVERTER_NAME, ASM_CONVERTER_VERSION
+from allotropy.parsers.chemometec_nucleoview.constants import (
+    DEFAULT_ANALYST,
+    DEFAULT_MODEL_NUMBER,
+    NUCLEOCOUNTER_SOFTWARE_NAME,
+)
+from allotropy.parsers.chemometec_nucleoview.nucleoview_reader import NucleoviewReader
+from allotropy.parsers.vendor_parser import VendorParser
+
+_PROPERTY_LOOKUP = {
+    "Dead (cells/ml)": TQuantityValueMillionCellsPerMilliliter,
+    "Estimated cell diameter (um)": TQuantityValueMicrometer,
+    "Live (cells/ml)": TQuantityValueMillionCellsPerMilliliter,
+    "Multiplication factor": TQuantityValueUnitless,
+    "Total (cells/ml)": TQuantityValueMillionCellsPerMilliliter,
+    "Viability (%)": TQuantityValuePercent,
+}
+
+
+def _get_value(data_frame: pd.DataFrame, row: int, column: str) -> Optional[Any]:
+    if column not in data_frame.columns:
+        return None
+    return data_frame[column][row]
+
+
+def get_property_from_sample(
+    data_frame: pd.DataFrame, row: int, property_name: str
+) -> Optional[Any]:
+    value = _get_value(data_frame, row, property_name)
+    if value is None:
+        return None
+
+    property_type = _PROPERTY_LOOKUP[property_name]
+
+    # if the porperty type is measured in million cells per ml convert cells per ml
+    if property_type == TQuantityValueMillionCellsPerMilliliter:
+        return property_type(value=float(value) / 1e6)
+
+    return property_type(value=float(value))
```

Comment:
is the float cast necessary here?

Reply:
yes, when I leave the cast out I end up with float values represented as strings in the final ASM output.  I think this is because I end up with a 2 column dataframe after pivot that roughly emulates a map or Series.  Since some of the rows have text in the "value" column I get text instead of floats.